recipes/omr1/pipeline/tir_solution_generation.py

```diff
@@ -0,0 +1,277 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from pathlib import Path
+
+import yaml
+
+from nemo_skills.pipeline.cli import generate, run_cmd, wrap_arguments
+
+
+def generate_solutions(input_file, output_dir, suffix, cluster, expname, extra_args="", **generate_kwargs):
+    run_after = f"{expname}-merge-data"  # this is launched in problem_generation.py
+
+    generate(
+        ctx=wrap_arguments(
+            f"++input_file={input_file} "
+            f"++prompt_config=generic/math "
+            f"++inference.temperature=0.7 "
+            f"{extra_args} "
+        ),
+        cluster=cluster,
+        output_dir=f"{output_dir}/generate-solutions-{suffix}",
+        expname=f"{expname}-generate-solutions-{suffix}",
+        run_after=run_after,
+        **generate_kwargs,
+    )
+
+
+def fill_majority_answer(output_dir, suffix, cluster, expname, extra_args="", **generate_kwargs):
+    run_after = f"{expname}-generate-solutions-{suffix}"
+
+    cmd = (
+        f'python -m nemo_skills.evaluation.aggregate_answers '
+        f'    ++input_dir={output_dir}/generate-solutions-{suffix} '
+        f'    ++output_dir={output_dir}/filled-majority-{suffix} '
+        f'    ++input_files="output-rs*.jsonl" '
+        f'    ++mode=fill '
+        f'    ++fill_is_correct=False '
+        f'    ++ignore_if_not_none=True '
+    )
+    run_cmd(
+        ctx=wrap_arguments(cmd),
+        cluster=cluster,
+        partition="cpu",  # change that if not available (ignored if running locally)
+        log_dir=f"{output_dir}/filled-majority-{suffix}/logs",
+        expname=f"{expname}-fill-majority-{suffix}",
+        run_after=run_after,
+    )
+
+
+def judge_answers(output_dir, suffix, cluster, expname, extra_args="", **generate_kwargs):
+    run_after = f"{expname}-fill-majority-{suffix}"
+
+    generate(
+        ctx=wrap_arguments(f"++input_dir={output_dir}/filled-majority-{suffix} {extra_args} "),
+        cluster=cluster,
+        generation_type="math_judge",
+        output_dir=f"{output_dir}/judged-generations-{suffix}",
+        expname=f"{expname}-judge-answers-{suffix}",
+        run_after=run_after,
+        **generate_kwargs,
+    )
+
+
+def preprocess_tir_generations(output_dir, suffix, cluster, expname, extra_args="", **generate_kwargs):
+    cmd = (
+        f"python /nemo_run/code/recipes/omr1/scripts/preprocess_tir_generations.py "
+        f"--input_files '{output_dir}/judged-generations-{suffix}/output-rs*.jsonl' "
+        f"--output_file {output_dir}/tir-preprocess-generations-{suffix}/preprocessed_output.jsonl "
+        f"--code_begin '```python' "
+        f"--code_end '```' "
+    )
+    run_cmd(
+        ctx=wrap_arguments(cmd),
+        cluster=cluster,
+        partition="cpu",
```

Comment:
don't use non-default partitions, but add a comment that it can be used if available
```suggestion
```